packages/react/src/SelectPanel/SelectPanel.features.stories.tsx

```diff
@@ -369,3 +369,70 @@ export const WithGroups = () => {
     />
   )
 }
+
+export const ModalVariant = () => {
+  const initialItems = [items[0], items[1]]
+  const [selected, setSelected] = React.useState<ItemInput[]>(initialItems)
+  const [filter, setFilter] = React.useState('')
+  const filteredItems = items.filter(item => item.text.toLowerCase().startsWith(filter.toLowerCase()))
+  const [open, setOpen] = useState(false)
+  const buttonRef = useRef<HTMLButtonElement>(null)
+
+  return (
+    <>
+      <h1>Multi Select Panel as Modal</h1>
+      <SelectPanel
+        variant="modal"
+        onCancel={() => setSelected(initialItems)}
```

Comment:
I wonder if it is doable/makes sense to handle the cancel in the component? 🤔 We can tae a call back in case there is anything else needs to be done after the cancel. Curious to hear what you think

Reply:
I would like that a lot. A combination of onSubmit and onCancel so that developers don't have to take care of transient state buuuuut that doesn't match what we have right now. `onSelectedChange` is fired immediately when an item is selected and the developer are expected to keep track of `selected` items and pass them back to SelectPanel 🤷 

I'd like to explore an API that abstracts the transient state but that is likely a breaking change, so only as a bonus at the end.

Reply:
>buuuuut that doesn't match what we have right now. onSelectedChange is fired immediately when an item is selected and the developer are expected to keep track of selected items and pass them back to SelectPanel 🤷

Yeah, I see! We don't have much control over the selected items, and it would probably violate the pattern if we try managing the state of the "selected" 

>I'd like to explore an API that abstracts the transient state but that is likely going to be a breaking change, so probably as a bonus task right at the end of the project

That would be interesting to explore! Yeah, no worries at all.